### PR TITLE
Node 502 skill crud operations

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
Create API to Create / Retrieve / Update / Delete Skills
[#502](https://github.com/nerubia/kh360-nodejs/issues/502)
FE: https://github.com/nerubia/kh360-react/pull/902

### CODE REVIEW

**Review by gborgonia-ir — APPROVED** [review]

File: src/services/skill-service.ts [hunk 1]
```diff
@@ -95,6 +95,10 @@ export const deleteById = async (id: number) => {
   if (skill === null) {
     throw new CustomError("Skill not found", 400)
   }
+  const relatedRecords = await SkillRepository.constrainedSkills(id)
+  if (relatedRecords.projectSkills.length > 0 || relatedRecords.projectMemberSkills.length > 0) {
```
Comment (emmaacha) [inline src/services/skill-service.ts:99]: Then do the checking here, if (existingProjectSkillCount > 0 || existingProjectMemberSkillCount > 0)

File: src/repositories/skill-repository.ts [hunk 2]
```diff
@@ -117,3 +117,21 @@ export const deleteById = async (id: number, skill_id: number) => {
     },
   })
 }
+
+export const constrainedSkills = async (id: number) => {
```
[the remark this answers is not in the document]
Comment (chadiegilnerubia) [reply src/repositories/skill-repository.ts]: Nice :3 thanks

File: src/services/skill-service.ts (hunk 1 above)
Comment (chadiegilnerubia) [reply src/services/skill-service.ts]: done :3

File: src/repositories/project-member-skill-repository.ts [hunk 3]
```diff
@@ -9,6 +9,12 @@ export const getById = async (id: number) => {
   })
 }
 
+export const countByFilters = async (where: Prisma.project_skillsWhereInput) => {
+  return await prisma.project_skills.count({
```
Comment (emmaacha) [inline src/repositories/project-member-skill-repository.ts:13]: this should be `project_member_skills`

**Review by emmaacha — CHANGES_REQUESTED** [review]

File: src/repositories/project-member-skill-repository.ts (hunk 3 above)
Comment (chadiegilnerubia) [reply src/repositories/project-member-skill-repository.ts:13]: oww mb

**Review by emmaacha — APPROVED** [review]

**Pull request merged**